.github/workflows/test.yml

```diff
@@ -2,35 +2,46 @@ name: Test
 
 on:
   pull_request:
-    paths:
-      - src/**
-      - tests/**
-      - 'phpunit.xml*'
-      - 'composer.*'
+    branches:
+      - 'master'
+    paths-ignore:
+      - '**.md'
   workflow_dispatch:
 
+concurrency:
+  group: '${{ github.workflow }}-${{ github.ref }}'
+  cancel-in-progress: true
+
 jobs:
   integration:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        php-version: ['5.6', '7.4', '8.0', '8.1']
+        php-version: ['5.6', '7.4', '8.0', '8.1', '8.2', '8.3']
+
+    name: Unit tests on PHP ${{ matrix.php-version }}
+
+    continue-on-error: ${{ matrix.php-version == '8.3' }}
+
     steps:
       - uses: actions/checkout@v3
-      
+
       - uses: shivammathur/setup-php@v2
         with:
-          tools: composer
           php-version: ${{ matrix.php-version }}
 
+      - name: Install Composer dependencies
+        uses: ramsey/composer-install@v2
+        with:
+          # Bust the cache at least once a month - output format: YYYY-MM.
+          custom-cache-suffix: $(date -u "+%Y-%m")
+
       - name: 'Validate composer.json'
         run: composer validate --strict
 
       - name: 'Validate PHP syntax'
         run: php -l src/db.php
 
-      - run: composer install
```

Comment:
Any reason why this was moved earlier? The validation and basic linting don't depend on the dependencies Composer installs and would be best to fail before more time is spent installing them potentially unnecessarily.

Reply:
I was planning on improving the overall matrix in a separate pr where these kinds of basic checks would be run first, and the test and other lint checks would then depend on that first job. 

I removed this step because the `ramsey/composer-install` action does the installation automatically. Not sure if I could pull these checks before the installation step with that action tho 🤔 I could slightly modify the test matrix to add these steps as a separate check like explained above.

P.S. Sorry for slower answer time I'm in the process of moving so I'm packing in parallel 😅

Reply:
Oh, I understand that the new action handles the install, my question was more about why we wouldn't keep that running in the same spot in the job. I don't see a benefit to running it before the validation steps. No rush, thanks!

Reply:
I pushed a change to keep this here for now, otherwise this should be good to go!